### PR TITLE
Configure Caching Guide V1

### LINKED ISSUE #9459: Add Configuring Clustering guide

### PR DESCRIPTION
**Pull request by DGuhr**
Closes #9459

### CODE REVIEW

File: docs/guides/src/main/server/caching.adoc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+The cache configuration is a regular https://infinispan.org/docs/stable/titles/configuring/configuring.html[Infinispan] configuration file.
+
+=== Overview: Cache types and Defaults
+The following table gives an overview of the specific caches Keycloak uses. These caches are configured in `conf/cache-ispn.xml`:
+
+|====
+|Cache name|Cache Type|Description
+|realms|Local|Cache persisted realm data
+|users|Local|Cache persisted user data
+|authorization|Local|Cache persisted authorization data
```
Comment (stianst) [inline docs/guides/src/main/server/caching.adoc:33]: Would mention that these are invalidated through events in the `work` cache

File: docs/guides/src/main/server/caching.adoc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+|===
+
+For the cloud vendor specific stacks to work, you have to provide additional dependencies to Keycloak. You can find more information, including Links to repositories providing the additional dependencies, in the https://infinispan.org/docs/dev/titles/embedding/embedding.html#jgroups-cloud-discovery-protocols_cluster-transport[official Infinispan documentation].
+
+To provide the dependencies to Keycloak, put the respective JAR in the `providers` directory and `build` Keycloak afterwards, using
+
+<@kc.build parameters="--cache-stack=<ec2|google|azure>"/>
+
+=== Securing cache communication
+The current Infinispan cache implementation can be secured by various security measures such as RBAC, ACLs and Transport stack encryption. Please refer to the https://infinispan.org/docs/dev/titles/security/security.html#[Infinispan security guide] for more information about securing cache communication.
```
Comment (stianst) [inline docs/guides/src/main/server/caching.adoc:93]:
```suggestion
The current Infinispan cache implementation should be secured by various security measures such as RBAC, ACLs and Transport stack encryption. Please refer to the https://infinispan.org/docs/dev/titles/security/security.html#[Infinispan security guide] for more information about securing cache communication.
```

**Review by stianst — COMMENTED** [review]
See comments

File: docs/guides/src/main/server/caching.adoc (hunk 1 above)
Comment (DGuhr) [reply docs/guides/src/main/server/caching.adoc:33]: Would it suffice to add a sentence around that in the paragraph below? Alternatively, I could split this up into two tables, one for the local ones with paragraph below, one for the distributed/replicated ones. wdyt?

File: docs/guides/src/main/server/caching.adoc [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+There are four distributed caches for sessions of authenticated users and services:
+
+* sessions
+* clientSessions
+* offlineSessions
+* offlineClientSessions
+
+These caches are used to save data about user sessions, which are valid for the length of a user’s browser session.
+
+.Fraud detection
```
Comment (stianst) [inline docs/guides/src/main/server/caching.adoc:52]:
```suggestion
.Password brute force detection
```

File: docs/guides/src/main/server/caching.adoc [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+=== Available transport stacks
+The following table shows transport stacks that are available without any further configuration than using the `--cache-stack` build option:
+|===
+|Stack name|Transport protocol|Discovery
+|tcp|TCP|MPING (uses UDP multicast).
+|udp|UDP|UDP multicast
+|kubernetes|TCP|DNS_PING
+|===
+
+=== Supported transport stacks
```
Comment (stianst) [inline docs/guides/src/main/server/caching.adoc:106]:
```suggestion
=== Additional transport stacks
```

File: docs/guides/src/main/server/caching.adoc [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+|===
+
+For the cloud vendor specific stacks to work, you have to provide additional dependencies to Keycloak. You can find more information, including Links to repositories providing the additional dependencies, in the https://infinispan.org/docs/dev/titles/embedding/embedding.html#jgroups-cloud-discovery-protocols_cluster-transport[official Infinispan documentation].
+
+To provide the dependencies to Keycloak, put the respective JAR in the `providers` directory and `build` Keycloak afterwards, using
+
+<@kc.build parameters="--cache-stack=<ec2|google|azure>"/>
+
+=== Securing cache communication
+The current Infinispan cache implementation can be secured by various security measures such as RBAC, ACLs and Transport stack encryption. Please refer to the https://infinispan.org/docs/dev/titles/security/security.html#[Infinispan security guide] for more information about securing cache communication.
```
Comment (stianst) [inline docs/guides/src/main/server/caching.adoc:124]:
```suggestion
The current Infinispan cache implementation should be secured by various security measures such as RBAC, ACLs and Transport stack encryption. Please refer to the https://infinispan.org/docs/dev/titles/security/security.html#[Infinispan security guide] for more information about securing cache communication.
```

**Review by stianst — APPROVED** [review]

**Pull request merged**